### DISCUSSION TITLE
How to setup extra advertisement_data (corret format) without bluer/bluez falling over?

### DISCUSSION THREAD

**Discussion by taurr** [bluez/bluer#124] [General]
For a reimplementation of an old service, we need to send Advertisements containing specific data.

* We need to set **both** local short name (type 0x08) **and** local complete name (type 0x09)
* The service UUID must be specified using type 0x02 (Incomplete list of uuids), and not 0x03 (complete list of uuids)
* we must be able to set the conn_interval_range (type 0x12)

When using `Bluer`, it seems only the complete name can be set, and the short one might be derived.

Trying to set e.g. type 0x12 as `advertisement_data` only results in an error: `Bluetooth operation failed: Failed to parse advertisement.`
Unfortunately the trace log is no better:
```plain
024-02-14T14:31:57.469997Z TRACE bluer::adv: /org/bluez/bluer/advertising/6e76378e21d242debcff251d8d7ed940: org.bluez.LEAdvertisement1.Data = Some({18: Variant([0, 1, 0, 6])})    
2024-02-14T14:31:57.471988Z TRACE start: bluer: DBus error org.bluez.Error.Failed: Failed to parse advertisement.  
```

If we just use the `btmgmt` tool, we can setup all the data we need. e.g.:
```shell
sudo btmgmt add-adv -d 05083645463402010603025DFE08FF03280000000010 -s 13094772756E64666F73534D4152542036454634 1
```

So the big questin is; can we even do this using `Bluer`? If so, how?

**Comment by surban** [comment]
Could you try doing the same using the `bluetoothctl` tool? BlueR and `bluetoothctl` are using `bluetoothd`, while `btmgmt` bypasses the bluetooth daemon and uses the Bluetooth management API.
